IA shuffle and edit of Contribute section (#13998)
PrefectHQ/prefect: docs/3.0rc/community/install-contribute.mdx

```
prefect dev kubernetes-manifest

```

To access the Prefect UI running in a Kubernetes cluster, use the `kubectl port-forward` command to forward a port on your local 
machine to an open port within the cluster. For example:

```
kubectl port-forward deployment/prefect-dev 4200:4200

```

This forwards port 4200 on the default internal loop IP for localhost to the Prefect server deployment.

To tell the local `prefect` command how to communicate with the Prefect API running in Kubernetes, set the `PREFECT_API_URL` 
environment variable:

```
export PREFECT_API_URL=http://localhost:4200/api

```

Since you previously configured port forwarding for the localhost port to the Kubernetes environment, you can interact 
with the Prefect API running in Kubernetes when using local Prefect CLI commands.

### Add database migrations

To make changes to a table, first update the SQLAlchemy model in `src/prefect/server/database/orm_models.py`. For example, 
to add a new column to the `flow_run` table, add a new column to the `FlowRun` model:

```
# src/prefect/server/database/orm_models.py

@declarative_mixin
class ORMFlowRun(ORMRun):
    """SQLAlchemy model of a flow run."""
    ...

>     new_column = Column(String, nullable=True) # <-- add this line

Reviewer: This line and next are showing up as problematic in GitHub. Is it looking okay in local deb server?

---
title: Install for development
---

Thanks for contributing to Prefect!

## Set up a development environment

> First, download the source code and install an editable version of the Python package:

Reviewer: ```suggestion
First, download the source code and install an editable copy of the Prefect Python package:
```

```python
# Clone the repository
git clone https://github.com/PrefectHQ/prefect.git
cd prefect

> # We recommend using a virtual environment

Reviewer: ```suggestion
# We recommend using a virtual environment. Below we use venv
```

python -m venv .venv
source .venv/bin/activate

# Install the package with development dependencies

pip install -e ".[dev]"

> # Setup pre-commit hooks for required formatting

Reviewer: ```suggestion
# Set up pre-commit hooks for required formatting
```

pre-commit install

```

If you don't want to install the pre-commit hooks, manually install the formatting dependencies with:

```
pip install $(./scripts/precommit-versions.py)

```

> Before a contribution is accepted, you need to run `ruff check` and `ruff format`.

Reviewer: ```suggestion
Run `ruff check` and `ruff format`.
```

After installation, you can run the test suite with `pytest`:

```
# Run all the tests
pytest tests

# Run a subset of tests

pytest tests/test_flows.py

```

<Tip>
**Build the Prefect UI**

If you intend to run a local Prefect server during development, first build the UI. 
See [UI development](#ui-development) for instructions.
</Tip>

## Prefect Code of Conduct

### Our Pledge

In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making 
participation in our project and our community a harassment-free experience for everyone. This is regardless of age, body size, 
disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, 
nationality, personal appearance, race, religion, or sexual identity and orientation.

### Our standards

> Examples of behavior that contributes to creating a positive environment include:

Reviewer: ```suggestion
Examples of behavior that contribute to creating a positive environment include:
```

Reviewer: I think "contributes" is matching "Examples of behavior", plural, as opposed to "behavior".

Reviewer 2: Oh, you're totally right. What do you think of dropping "Examples of" such that it reads "Behavior that contributes.."

Reviewer: Sounds good to me.

### Enforcement

Report instances of abusive, harassing, or otherwise unacceptable behavior by contacting Chris White at 
[chris@prefect.io](mailto:chris@prefect.io). All complaints are reviewed and investigated. Each complaint will receive a 
response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain 
confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted 
separately.

Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent 
repercussions, as determined by other members of the project's leadership.

### Attribution

This Code of Conduct is adapted from the [Contributor Covenant](https://www.contributor-covenant.org/), version 1.4, 
available at [https://www.contributor-covenant.org/version/1/4/code-of-conduct.html](https://www.contributor-covenant.org/version/1/4/code-of-conduct.html)

For answers to common questions about this code of conduct, see [https://www.contributor-covenant.org/faq](https://www.contributor-covenant.org/faq)

> ## Developer Tooling 

Reviewer: ```suggestion
## Developer tooling 
```

The Prefect CLI provides several helpful commands to aid development.

Start all services with hot-reloading on code changes (requires installation of UI dependencies):

```
prefect dev start
```

Start a Prefect API that reloads on code changes:

```
prefect dev api
```

> Start a Prefect worker that reloads on code changes:

Reviewer: ```suggestion
```

> ```

Reviewer: ```suggestion
```

> prefect dev agent

Reviewer: ```suggestion
```

> ```

Reviewer: ```suggestion
```

### UI development

Developing the Prefect UI requires installation of [npm](https://github.com/npm/cli).

Start a development UI that reloads on code changes:

```
prefect dev ui
```

Build the static UI (the UI served by `prefect server start`):

```
prefect dev build-ui
```

> ### Docs Development

Reviewer: I'll update this in a fast-follow PR.
